### PR TITLE
Fixed focusing a widget after blurring editor for edge

### LINKED ISSUE #1458: After blurring editor it takes 2 clicks to focus a widget
## Are you reporting a feature request or a bug?

Bug

## Provide detailed reproduction steps (if any)

1. Open any [CKE4 instance with Enhanced Image plugin](http://cdn.ckeditor.com/4.8.0/full-all/samples/old/image2/image2.html).
1. Focus the image widget by clicking it.
1. Blur the editor, by clicking outside of the editor.
1. Click image once again to focus it.

### Expected result

Image should be focused.

### Actual result

Image is not focused. Only next subsequent click will focus it.

## Other details

This is a regression introduced in 4.7.0 (it might happen it was not present in previous edge releases so we didn't see it).

It's a general widget issue, not limited to the Enhanced Image plugin.

* Browser: Edge
* OS: Windows 10
* CKEditor version: 4.8.0

### PR DESCRIPTION
**Pull request by jacekbogdanski**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests) and
[how to create tests](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [ ] Unit tests
- [x] Manual tests

## What changes did you make?

Added condition to focus editor for [CKEDITOR.plugins.copyformatting.state#applyFormatting](https://docs.ckeditor.com/ckeditor4/docs/#!/api/CKEDITOR.plugins.copyformatting.state-event-applyFormatting) event only when client browser is IE .
Closes #1458

### CODE REVIEW

File: tests/plugins/copyformatting/manual/focus.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	<div style="float:right; padding:5px; width:210px"><a href="http://ckeditor.com/"><img alt="Focus jumps to the top of the editor" src="http://c.cksource.com/a/1/img/demo/brownie.jpg" style="height:133px; width:200px" /></a></div>
+
+	<p>&nbsp;</p>
+
+	<p>&nbsp;</p>
+</textarea>
+
+<script>
+	CKEDITOR.replace( 'editor1' );
```
Comment (f1ames) [inline tests/plugins/copyformatting/manual/focus.html]: Again, from what I understand this was the Edge only fix, what was the reason of removing
```
if ( !CKEDITOR.env.ie ) {		
 	bender.ignore();		
 }
```
ignore from this test? Does the changes somehow affect other browsers?

File: plugins/copyformatting/plugin.js [hunk 2]
```diff
@@ -414,7 +414,7 @@
 			},
 
 			applyFormatting: {
-				editorFocus: false,
+				editorFocus: CKEDITOR.env.ie && !CKEDITOR.env.edge ? false : true,
```
[the remark this answers is not in the document]
Comment (jacekbogdanski) [reply plugins/copyformatting/plugin.js]: No, it should be false for an IE. It is old fix where IE with editor focus will scroll up if an image is at the bottom of a scrollable container bdbd071.

File: tests/plugins/copyformatting/manual/focus.html (hunk 1 above)
Comment (jacekbogdanski) [reply tests/plugins/copyformatting/manual/focus.html]: The problem was with IE, but older fix corrupted Edge, so I thought it's good idea to test all other browsers.

File: plugins/copyformatting/plugin.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (f1ames) [reply plugins/copyformatting/plugin.js]: Ok, I see, after F2F talk, the reason was to rollback the `editorFocus: false` change as initially it should be applied only for IEs, but was applied for all browsers (see https://dev.ckeditor.com/ticket/16845). So now `editorFocus: false` should be only used for IEs.
That was also a reason for modifying manual test (which is answer for https://github.com/ckeditor/ckeditor-dev/pull/1466#discussion_r162580742).

[the remark this answers is not in the document]
Comment (jacekbogdanski) [reply plugins/copyformatting/plugin.js]: Sorry for misleading information about unit tests. As we talked F2F there is no easy way to test this issue directly in the code, so I will skip this part and use manual tests.

File: tests/plugins/copyformatting/manual/edgefocus.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+@bender-tags: bug, 4.9.0, copyformatting, 1458
```
Comment (f1ames) [inline tests/plugins/copyformatting/manual/edgefocus.md]: Please update tag to `4.9.1`.

File: tests/plugins/copyformatting/manual/edgefocus.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+@bender-ui: collapsed
+@bender-ckeditor-plugins: copyformatting,toolbar,wysiwygarea,floatingspace,elementspath,image2,widget,justify,div
```
Comment (f1ames) [inline tests/plugins/copyformatting/manual/edgefocus.md]: You could easily run this test with only `copyformatting,toolbar,wysiwygarea,image2`. Noe need for additional plugins.

File: tests/plugins/copyformatting/manual/edgefocus.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+## Test Scenario
+
+1. Make sure the editor is not focused.
+2. Select the image on the left by clicking it.
```
Comment (f1ames) [inline tests/plugins/copyformatting/manual/edgefocus.md]: > by clicking **on** it.

File: tests/plugins/copyformatting/manual/edgefocus.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+3. Remove selection from the image by clicking outside of the editor.
```
Comment (f1ames) [inline tests/plugins/copyformatting/manual/edgefocus.md]: > outside ~~of~~ the editor.

File: tests/plugins/copyformatting/manual/focus.html [hunk 7]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (f1ames) [inline tests/plugins/copyformatting/manual/focus.html]: Is this blank line really necessary?

File: tests/plugins/copyformatting/manual/focus.md [hunk 8]
```diff
@@ -1,4 +1,4 @@
-@bender-tags: bug, 4.7.0, trac16845, copyformatting
+@bender-tags: bug, 4.7.0, 4.9.0, trac16845, copyformatting, 1458
 @bender-ui: collapsed
 @bender-ckeditor-plugins: copyformatting,toolbar,wysiwygarea,floatingspace,elementspath,image2,widget,justify,div
```
Comment (f1ames) [inline tests/plugins/copyformatting/manual/focus.md]: Same here, you only need `copyformatting,toolbar,wysiwygarea,image2,justify,div` (I know this tests was not added by this PR, but it is a good opportunity to fix it:D).

**Comment by f1ames** [comment]
Should be merged to `master` after `4.9.0` release.

**Comment by mlewand** [comment]
Retargetting into a `next` branch (4.10.0 release) as current major is frozen for the 4.9.0 release.

**Pull request merged**